File: TUnit.TestProject/ExperimentalTests.cs
using System.Diagnostics.CodeAnalysis;

namespace TUnit.TestProject;

public class ExperimentalTests
{
    [Experimental("Blah")]
    [Test]
    public void SynchronousTest()
    {
        // Dummy method
    }

    [Experimental("Blah")]
    [Test]
    public async Task AsynchronousTest()
    {
        await Task.CompletedTask;
    }
}

File: TUnit.TestProject.FSharp/ExperimentalTests.fs
namespace TUnit.TestProject.FSharp

open System.Threading.Tasks
open TUnit.Core

// F# equivalent of ExperimentalTests.cs

type ExperimentalTests() =
    [<Experimental("Blah")>]
    [<Test>]
    member _.SynchronousTest() = ()

    [<Experimental("Blah")>]
    [<Test>]
    member _.AsynchronousTest() : Task = Task.CompletedTask
